### PR TITLE
[DDW-1174] Add custom dimension to track legacy and hardware wallets

### PR DESCRIPTION
**Pull request by marcin-mazurek**
This PR adds additional dimensions that will help us understand the percentage of users who still use legacy Byron wallets and usage of hardware wallets.

## Screenshots

n/a

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1670590862190569)
- [Jira ticket](https://input-output.atlassian.net/browse/DDW-1174)

---

### Basics

- [x] PR assigned to the PR author(s)
- [x] `input-output-hk/daedalus-dev` and `input-output-hk/daedalus-qa` assigned as PR reviewers
- [ ] ~If there are UI changes, Alexander Rukin assigned as an additional reviewer~
- [ ] ~All visual regression testing has been reviewed (assign `run Chromatic` label to PR to trigger the run)~
- [x] PR has appropriate labels (`release-vNext`, `feature`/`bug`/`chore`, `WIP`)
- [x] PR link is added to a Jira ticket, ticket moved to In Review
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [ ] ~PR contains screenshots (in case of UI changes)~
- [x] CHANGELOG entry has been added to the top of the appropriate section (_Features_, _Fixes_, _Chores_) and is linked to the correct PR on GitHub
- [ ] ~There are no missing translations (running `yarn manage:translations` produces no changes)~
- [ ] ~Text changes are proofread and approved (Jane Wild / Amy Reeve)~
- [ ] ~Japanese text changes are proofread and approved (Junko Oda)~
- [ ] ~Storybook works and no stories are broken (`yarn storybook`)~
- [ ] ~In case of dependency changes `yarn.lock` file is updated~

### Code Quality

- [x] Important parts of the code are properly commented and documented
- [x] Code is properly typed with typescript types
- [x] React components are split-up enough to avoid unnecessary re-renderings
- [x] Any code that only works in main process is neatly separated from components

### Testing

- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review

- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: source/renderer/app/analytics/getCustomDimensions.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { formattedBytesToSize } from '../utils/formatters';
+import AdaApi from '../api/api';
+
+const booleanToText = (flag: boolean) => (flag ? 'yes' : 'no');
+
+export const getCustomDimensions = async (
+  environment: Environment,
+  adaApi: AdaApi
+) => {
+  const usesByronWallets = (await adaApi.getWallets()).some(
```
Comment (lucas-barros) [inline source/renderer/app/analytics/getCustomDimensions.ts:20]: Small suggestion for single wallet request
```
const wallets = await adaApi.getWallets()
const usesHardwareWallets = wallets.some(
  (wallet) => wallet.isHardwareWallet
);
```

**Review by lucas-barros — APPROVED** [review]

File: source/renderer/app/analytics/getCustomDimensions.ts (hunk 1 above)
Comment (marcin-mazurek) [reply source/renderer/app/analytics/getCustomDimensions.ts]: Good catch, not sure why I missed it. Fixed.

**Review by danielmain — APPROVED** [review]
Nice you did a refactoring for custom dimensions, great Job @marcin-mazurek

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 - Implemented CPU name grouping mechanism for analytics ([PR 3089](https://github.com/input-output-hk/daedalus/pull/3089))
 - Implemented rounding RAM size in analytics ([PR 3088](https://github.com/input-output-hk/daedalus/pull/3088))
+- Added custom dimension to track legacy and hardware wallets ([PR 3091](https://github.com/input-output-hk/daedalus/pull/3091))
```
Comment (tomislavhoracek) [inline CHANGELOG.md:9]: @marcin-mazurek could you move the log to the very top please 🙏

**Review by tomislavhoracek — APPROVED** [review]
💯

File: CHANGELOG.md (hunk 2 above)
Comment (marcin-mazurek) [reply CHANGELOG.md]: Done

**Review by ManusMcCole — APPROVED** [review]
Looks good 👍

**Pull request merged**